boards/common/nrf52xxxdk/Makefile.include

```diff
@@ -15,18 +15,32 @@ PORT_LINUX ?= /dev/ttyACM0
 PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.usbmodem*)))
 include $(RIOTMAKE)/tools/serial.inc.mk
 
-# setup JLink for flashing
-export JLINK_DEVICE := nrf52
-
 # The following configuration is dependencies specific
 # but they are resolved later
 # Hack to know now if 'nordic_softdevice_ble' is used
 include $(RIOTBOARD)/$(BOARD)/Makefile.dep
 
-# special options when using SoftDevice
-ifneq (,$(filter nordic_softdevice_ble,$(USEPKG)))
-  export JLINK_PRE_FLASH := erase\nloadfile $(BINDIR)/softdevice.hex
-  export FLASH_ADDR := 0x1f000
-  export LINKER_SCRIPT ?= $(RIOTCPU)/$(CPU)/ldscripts/$(CPU_MODEL)_sd.ld
+PROGRAMMER ?= jlink
+ifeq (jlink,$(PROGRAMMER))
+  # setup JLink for flashing
+  export JLINK_DEVICE := nrf52
+
+  # special options when using SoftDevice
+  ifneq (,$(filter nordic_softdevice_ble,$(USEPKG)))
+    export JLINK_PRE_FLASH := erase\nloadfile $(BINDIR)/softdevice.hex
+    export FLASH_ADDR := 0x1f000
+    export LINKER_SCRIPT ?= $(RIOTCPU)/$(CPU)/ldscripts/$(CPU_MODEL)_sd.ld
+  endif
+  include $(RIOTMAKE)/tools/jlink.inc.mk
+else ifeq (openocd,$(PROGRAMMER))
+  # setup OpenOCD for flashing. Version 0.10 of OpenOCD doesn't contain support
+  # for nrf52dk and nrf52840dk boards. To use OpenOCD with these a version
+  # build from source (master > 2018, August the 13rd) is required.
```

Comment:
I would put this info in Doxygen, as users will be more likely to see it there. Feel free to use https://github.com/RIOT-OS/RIOT/pull/10890/commits/740814a62cfd7f5a62c03958783e2b2e99015016